### PR TITLE
Fix build warnings (treated as errors)

### PR DESCRIPTION
**Pull request by dholbert**
These commits fix https://github.com/rr-debugger/rr/issues/3011 and https://github.com/rr-debugger/rr/issues/3012 and let my build complete successfully.

### CODE REVIEW

File: src/test/strict_priorities.c [hunk 1]
```diff
@@ -38,6 +38,6 @@ int main(void) {
   main_thread_done = 1;
 
   atomic_puts("EXIT-SUCCESS");
-
+  (void)dummy; /* Avoid -Wunused-but-set-variable build warning. */
```
Comment (Keno) [inline src/test/strict_priorities.c:41]: This is probably a real bug in this test. The compiler will delete the above loop if this variable is unused. Could be fixed with either a volatile or some sort of asm memory call.

Comment (dholbert) [reply src/test/strict_priorities.c:41]: Interesting! I admit I didn't actually do serious analysis of what the test is trying to do here.

If the test does in fact need fixing beyond this warning-silencing-bandaid, I'll leave that to someone who's got some familiarity with rr internals/development/testing (i.e. not me).

**Comment by rocallahan** [comment]
I rewrote your first patch to use 'volatile' and landed both patches.

**Comment by dholbert** [comment]
Thanks!

**Pull request closed without merging**